net/dns/manager_openbsd.go

```diff
@@ -4,8 +4,72 @@
 
 package dns
 
-import "tailscale.com/types/logger"
+import (
+	"bytes"
+	"fmt"
+	"os"
 
-func NewOSConfigurator(logger.Logf, string) (OSConfigurator, error) {
-	return newDirectManager(), nil
+	"tailscale.com/types/logger"
+)
+
+type kv struct {
+	k, v string
+}
+
+func (kv kv) String() string {
+	return fmt.Sprintf("%s=%s", kv.k, kv.v)
+}
+
+func NewOSConfigurator(logf logger.Logf, interfaceName string) (OSConfigurator, error) {
+	return newOSConfigurator(logf, interfaceName,
+		newOSConfigEnv{
+			rcIsResolvd: rcIsResolvd,
+			fs:          directFS{},
+		})
+}
+
+// newOSConfigEnv
+type newOSConfigEnv struct {
+	fs          wholeFileFS
+	rcIsResolvd func(resolvConfContents []byte) string
+}
+
+func newOSConfigurator(logf logger.Logf, interfaceName string, env newOSConfigEnv) (ret OSConfigurator, err error) {
+	var debug []kv
+	dbg := func(k, v string) {
+		debug = append(debug, kv{k, v})
+	}
+	defer func() {
+		if ret != nil {
+			dbg("ret", fmt.Sprintf("%T", ret))
+		}
+		logf("dns: %v", debug)
+	}()
+
+	bs, err := env.fs.ReadFile(resolvConf)
+	if os.IsNotExist(err) {
+		dbg("rc", "missing")
+		return newDirectManager(), nil
+	}
+	if err != nil {
+		return nil, fmt.Errorf("reading /etc/resolv.conf: %w", err)
+	}
+
+	switch env.rcIsResolvd(bs) {
+	case "resolvd":
+		dbg("resolvd", "yes")
+		return newResolvdManager(logf, interfaceName)
+	default:
+		dbg("resolvd", "missing")
+		return newDirectManager(), nil
+	}
+}
+
+func rcIsResolvd(resolvConfContents []byte) string {
```

Comment:
I'd make this func return a bool if it's making a binary decision. I believe you patterned this on linux's resolvOwner, which has to return an N-way decision between all the different managers. OpenBSD is mercifully simpler :)